.github/workflows/test.yml

```diff
@@ -31,88 +30,33 @@ jobs:
           java-version: '17'
           cache: 'gradle'
 
-      - name: AVD cache
-        uses: actions/cache@v3
-        id: avd-cache
-        with:
-          path: |
-            ~/.android/avd/*
-            ~/.android/adb*
-          key: aosp-${{ matrix.emulator_type }}-${{ matrix.android_target }}-${{ runner.os }}
-
       - name: Build Release APK
         run: ./gradlew :app:assembleRelease
 
-      - name: Create AVD snapshot
-        if: steps.avd-cache.outputs.cache-hit != 'true'
-        uses: reactivecircus/android-emulator-runner@v2
-        with:
-          api-level: ${{ matrix.android_target }}
-          target: ${{ matrix.emulator_type }}
-          arch: x86_64
-          force-avd-creation: false
-          emulator-options: -writable-system -no-snapshot-load -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
-          disable-animations: true
-          script: |
-            ./app/development/scripts/provision_emulator.sh "test" "system-images;android-${{ matrix.android_target }};${{ matrix.emulator_type }};x86_64"
-            echo "Generated AVD snapshot for caching."
-
       - name: Assemble tests
         run: ./gradlew :app:assembleAndroidTest
 
       - name: Run tests
-        uses: reactivecircus/android-emulator-runner@v2
+        uses: Wandalen/wretry.action@v1.3.0
```

Comment:
we need to retry 3 times? Are things this flaky? do we know where the flakyness comes from?

Reply:
I'm not sure if we do, just leaving this for now. If we see consistency we can remove it, if not we can fix the flakiness.